docs/source/Objective_Functions/Objective_Function.md

```diff
@@ -1,25 +1,217 @@
+
+
+```
+from tq42.channel import Ask, Tell
+from tq42.client import TQ42Client
+
+with TQ42Client() as client:
+  objective_func_channel = await Channel.create(client=client)
+  local_opt_channel = await Channel.create(client=client)
+  
+  tetra_opt_parameters = {
+      "dimensionality": 2,
+      "iteration_number": 2,
+      "maximal_rank": 4,
+      "points_number": 1,
+      "quantization": False,
+      "tolerance": 0.0010000000474974513,
+      "lower_limits": [0, 0],
+      "upper_limits": [9, 9],
+      "grid": [10, 10],
+       "objective_function_channel_id": objective_func_channel.id
+       #local_optimizer_channel_id parameter is optional
+      "local_optimizer_channel_id": local_opt_channel.id
+  }
+```
+
+### Example Using HTTPS Endpoint
+Below is an example of how to set up TetraOpt parameters using an HTTPS endpoint. Notice the `objective_function` and `local_optimizer` parameters:
+
+```
+from tq42.client import TQ42Client
+
+tetra_opt_parameters = {
+    "dimensionality": 2,
+    "iteration_number": 2,
+    "maximal_rank": 4,
+    "points_number": 1,
+    "quantization": False,
+    "tolerance": 0.0010000000474974513,
+    "lower_limits": [0, 0],
+    "upper_limits": [9, 9],
+    "grid": [10, 10],
+    'objective_function':'http://34.32.169.11:8000/test_func_eval/Ackley/',
+     #local_optimizer parameter is optional  
+    "local_optimizer": "http://34.32.169.11:8000/local_optimization/Ackley/",
+    "polling": {"initial_delay": 1.0, "retries": 100, "delay": 1.0, "backoff_factor": 1.1}
+}
+```
+
+### 1. To use the communications channel:
+The communication channel uses the following classes to send and receive information:
+
+1. `Ask`
+2. `Tell`
+3. `Channel`
+
+
+You can import these classes from the tq42 library by:
+
+`from tq42.channel import Channel, Ask, Tell`
+
+
+####  Below are the Python types for the Ask and Tell parameters you need to know:
+  - `ask.headers` -> list of strings
+  - `ask.parameter.values` -> list of floats
+  - `tell.results` -> list of floats
+  - `tell.candidates.values` -> list of floats
+  
+
+
+#### The `Ask` Class
+An example of an `Ask` object values passed by TetraOpt:
+```
+{
+    "parameters": [
+        {"values": [1, 3]},
+        {"values": [2, 7]}
+    ],
+    "headers": ["h0", "h1"]
+}
+```
+
+#### The `Tell` Class 
+An example of `Tell` object values for an objective and local optimization function:
+```
+{
+    "parameters": [
+        {"values": [1, 3]},
+        {"values": [2, 7]}
+    ],
+    "headers": ["h0", "h1"],
+    "results": [6.59359884, 7.86938667],
+    #parameter candidates is only used for local optimization function
+    "candidates": [
+        {'values': array([0., 0.])},
+        {'values': array([-8.18565023e-09, -8.18565023e-09])},
+        {'values': array([5.02359648e-08, 5.02359648e-08])}, 
+        {'values': array([9., 9.])}
+    ]
+}
+```
+
+### Implementing Objective and Local Optimization Functions
+#### Objective Function Callback Example
```

Comment:
Can we remove this example completely from the documentation please? It is literally a copy paste of all parts of the jupyter notebook. If we keep things around let's do it only in one place so we do not have to update both places. Just reference the notebook for an example.